gulpfile.js

```diff
@@ -0,0 +1,103 @@
+var argv = require('yargs')
+    .usage(
+      'Usage: $0 --appengine [Path to app engine]' +
+      '--clear_datastore [clear all data storage]' +
+      '--enable_sendmail [send emails]' +
+      '--minify [minify only in production]').argv;
+var gulpStartGae = require('./scripts/gulp-start-gae-devserver');
+var manifest = require('./manifest.json');
+
+var serveCmd = argv.appengine;
+var params = {
+  admin_host: '0.0.0.0',
+  admin_port: 8000,
+  host: '0.0.0.0',
+  port: 8181
+};
+
+var isMinified = (argv.minify == 'True');
+var dependencies = manifest.dependencies.thirdPartyStatic;
+var cssFilesPath = [];
+var fontFolderPath = [];
+var cssBackgroundPath = [];
+for (var dependencyId in dependencies) {
+  var dependency = dependencies[dependencyId];
+  if (dependency.hasOwnProperty('cssFiles')) {
+    var cssFiles = dependency.cssFiles;
+    for (var files in cssFiles) {
+      cssFilesPath.push(path.join('third_party', 'static',
+      dependency.targetDirPrefix + dependency.version,
+      cssFiles[files]));
+    }
+  }
+  if (dependency.hasOwnProperty('fontsPath')) {
+    var fontPrefix = '*.{eot,woff2,ttf,woff,eof,svg}';
+    fontFolderPath.push(path.join('third_party', 'static',
+      dependency.targetDirPrefix + dependency.version,
+      dependency.fontsPath, fontPrefix));
+  }
+  if (dependency.hasOwnProperty('cssBackgroundPath_image')) {
+    var backgroundImage = dependency.cssBackgroundPath_image;
+    for (var image in backgroundImage) {
+      cssBackgroundPath.push(path.join('third_party', 'static',
+        dependency.targetDirPrefix + dependency.version,
+        backgroundImage[image]));
+    }
+  }
+}
+
+if (argv.clear_datastore) {
+  params.clear_datastore = true;
+}
+
+if (argv.enable_sendmail) {
+  params.enable_sendmail = argv.enable_sendmail;
+}
+
+var cssDir = path.join(
+  'third_party', 'generated', isMinified ? 'prod' : 'dev', 'css');
+gulp.task('css', function() {
+  gulp.src(cssFilesPath)
+    .pipe(isMinified ? minifyCss() : gutil.noop())
+    .pipe(concat('third_party.css'))
+    .pipe(gulp.dest(cssDir));
+});
+
+gulp.task('copyFonts', function() {
+  gulp.src(fontFolderPath)
+  .pipe(gulp.dest(path.join(
+    'third_party', 'generated', isMinified ? 'prod' : 'dev', 'fonts')));
+});
+
+gulp.task('copyCssBackgroundImages', function() {
+  gulp.src(cssBackgroundPath)
+  .pipe(gulp.dest(cssDir));
+});
+
+gulp.task('gulpStartGae', function() {
+  gulp.src('app.yaml')
+    .pipe(gulpStartGae(serveCmd, [], params));
+});
+
+// This takes all functions  that are required for the build
+// e.g css, Js and Images
+gulp.task('build', ['css', 'copyFonts', 'copyCssBackgroundImages']);
+
+gulp.slurped = false;
+gulp.task('watch', function() {
+  if (!gulp.slurped) {
+    gulp.watch('gulpfile.js', ['build']);
+    gulp.watch(cssFilesPath, ['css']);
+    gulp.watch('manifest.json', ['build']);
+    gulp.slurped = true;
+  }
+});
+
+// This takes all default task which must be run when start.sh is started.
```

Comment:
Is there anywhere you're doing the following check: the files have already been generated, so don't bother to regenerate them?